v2/internal/util/predicates/name_predicate.go

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package predicates
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// MakeNamePredicate creates a predicate that matches resources with a specific name only
+func MakeNamePredicate(names ...string) predicate.Predicate {
+	return makeObjectPredicate(func(_ client.Object, new client.Object) bool {
```

Comment:
Seeing a parameter named `new` throws me a bit - very used to seeing that as a keyword. Is this the best name? #veryminor